test/new-e2e/tests/remote-config/utils_test.go

```diff
@@ -8,37 +8,91 @@ package remoteconfig
 import (
 	_ "embed"
 	"fmt"
-	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/components"
-	"github.com/stretchr/testify/assert"
+	"os"
+	"path"
+	"strings"
 	"testing"
 	"time"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/components"
+	"github.com/stretchr/testify/assert"
 )
 
 // assertLogsEventually will verify that a given `agentName` component's logs contain a pattern.
 // It will continually retry until the `expectedLogPattern` is found or the `maxRetries` is reached,
 // waiting `retryInterval` between each attempt.
 // If the `expectedLogPattern` is not found or an error occurs, the calling test will fail.
-func assertLogsEventually(t *testing.T, rh *components.RemoteHost, agentName string, expectedLogPattern string, waitFor time.Duration, tick time.Duration) {
+func assertAgentLogsEventually(t *testing.T, rh *components.RemoteHost, agentName string, expectedLogs []string, waitFor time.Duration, tick time.Duration) {
 	t.Helper()
+	foundLogs := make(map[string]bool, len(expectedLogs))
+	missingLogs := make([]string, len(expectedLogs))
+	// initially all logs are missing
+	copy(missingLogs, expectedLogs)
+	remoteLogsPath := fmt.Sprintf("/var/log/datadog/%s.log", agentName)
+	t.Logf("looking for logs in %s", remoteLogsPath)
 	assert.EventuallyWithTf(t, func(c *assert.CollectT) {
-		output, err := rh.Execute(fmt.Sprintf("cat /var/log/datadog/%s.log", agentName))
-		if assert.NoError(c, err) {
-			assert.Contains(c, output, expectedLogPattern)
+		// read agent logs
+		agentLogs, err := readRemoteFile(rh, remoteLogsPath)
+		if !assert.NoError(c, err) {
+			return
+		}
+		for _, log := range missingLogs {
+			if strings.Contains(agentLogs, log) {
+				t.Logf("found log: %s", log)
+				foundLogs[log] = true
+			}
 		}
-	}, waitFor, tick, "failed to find log with pattern `%s`", expectedLogPattern)
+		// reset missing logs
+		missingLogs = make([]string, 0, len(expectedLogs))
+		for _, log := range expectedLogs {
+			if _, ok := foundLogs[log]; ok {
+				continue
+			}
+			missingLogs = append(missingLogs, log)
+		}
+		assert.Empty(c, missingLogs, "still missing logs")
+		t.Logf("missing logs:\n[%s]", strings.Join(missingLogs, ","))
+	}, waitFor, tick, "failed finding logs in agent")
 }
 
 // mustCurlAgentRcServiceEventually will curl the remote config service's endpoint to get tracer
 // configurations every `tick` until either it is successful (in which case it will return the
 // output of the curl command), or the `waitFor` duration is reached (in which case it will
 // fail the calling test).
-func mustCurlAgentRcServiceEventually(t *testing.T, rh *components.RemoteHost, payload string, waitFor time.Duration, tick time.Duration) string {
+func assertCurlAgentRcServiceContainsEventually(t *testing.T, rh *components.RemoteHost, payload string, expectedKeys []string, waitFor time.Duration, tick time.Duration) string {
```

Comment:
As this method is really similar to the previous one, could they share a common part?

Reply:
I would let remote-config handle it. These are internal only functions